gammapy/detect/lima.py

```diff
@@ -2,13 +2,68 @@
 import copy
 import logging
 import numpy as np
+from astropy.coordinates import Angle
+from astropy.convolution import Tophat2DKernel
 from gammapy.stats import significance, significance_on_off
+from gammapy.cube import MapDataset, MapDatasetOnOff
 
-__all__ = ["compute_lima_image", "compute_lima_on_off_image"]
+__all__ = [
+    "SignificanceMapEstimator",
+    "compute_lima_image",
+    "compute_lima_on_off_image",
+]
 
 log = logging.getLogger(__name__)
 
 
+class SignificanceMapEstimator:
+    """Computes correlated excess, significance for MapDatasets
+
+
+    Parameters
+    ----------
+    correlation_radius : ~astropy.coordinate.Angle
+        correlation radius to use
+    """
+
+    def __init__(self, correlation_radius):
+        self.radius = Angle(correlation_radius)
+
```

Comment:
Maybe attach the existing `compute_lima_image` methods as `staticmethod` on the `LiMaMapEstimator`? Then all the functionality is bundled in a single name-space? Like `LiMaMapEstimator.estimate_maps(counts, background, kernel)` and `LiMaMapEstimator.estimate_maps_on_off(counts, counts_off, acceptance, acceptance_off, kernel)`

Reply:
Done